### PR TITLE
Merge feature/IESP-139

### PR DESCRIPTION
**Pull request by angelaantoniaceaa**
Changed presentation logic of the service for searching for producer names

### CODE REVIEW

**Review by nttdata-rtorsoli — DISMISSED** [review]

File: src/main/java/it/pagopa/interop/probing/probingapi/service/impl/ProducerServiceImpl.java [hunk 1]
```diff
@@ -14,14 +16,21 @@ public class ProducerServiceImpl implements ProducerService {
...
-  }
+    SearchProducerNameBEResponse response =
+        producerClient.getProducers(limit, offset, producerName).getBody();
 
+    return !CollectionUtils.isEmpty(response.getContent())
+        ? response.getContent().stream().map(name -> {
+          return SearchProducerNameResponse.builder().label(name).value(name).build();
+        }).toList()
+        : List.of();
+  }
```
Comment (galales) [inline src/main/java/it/pagopa/interop/probing/probingapi/service/impl/ProducerServiceImpl.java]: Am I missing something or it equivalent to this?
```suggestion
    return response.getContent().stream().map(name -> {
          return SearchProducerNameResponse.builder().label(name).value(name).build();
        }).toList();
  }
```

`response.getContent()` should never be null, right?

Comment (angelaantoniaceaa) [reply src/main/java/it/pagopa/interop/probing/probingapi/service/impl/ProducerServiceImpl.java]: Yes, `response.getContent()` should never be null, but it's a check I added for better reliability. I'll redo it if you want

Comment (galales) [reply src/main/java/it/pagopa/interop/probing/probingapi/service/impl/ProducerServiceImpl.java]: if `operations` returns an empty response it means that there is a bug in that service, there is no need for all the callers to handle responses not in the specs

File: src/main/java/it/pagopa/interop/probing/probingapi/mapping/dto/impl/SearchProducerNameBEResponse.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class SearchProducerNameBEResponse implements Response {
+
+  private static final long serialVersionUID = 1L;
+
+  @JsonProperty("content")
+  @NotNull(message = "must not be null")
+  private List<String> content;
```
Comment (nttdata-rtorsoli) [inline src/main/java/it/pagopa/interop/probing/probingapi/mapping/dto/impl/SearchProducerNameBEResponse.java:22]: But it is not preferrable Collection`<T>`?

**Review by nttdata-rtorsoli — APPROVED** [review]

**Review by galales — APPROVED** [review]

**Pull request merged**